lib/hls/hls_parser.js

```diff
@@ -4179,15 +4214,50 @@ shaka.hls.HlsParser = class {
       return null;
     }
 
+    const keyUris = shaka.hls.Utils.constructSegmentUris(
+        getUris(), drmTag.getRequiredAttrValue('URI'), variables);
+
+    const requestType = shaka.net.NetworkingEngine.RequestType.KEY;
+    const request = shaka.net.NetworkingEngine.makeRequest(
+        keyUris, this.config_.retryParameters);
+    const keyResponse =
+        await this.makeNetworkRequest_(request, requestType);
+
+    const key = shaka.util.Uint8ArrayUtils.toHex(keyResponse.data);
+
     // NOTE: The ClearKey CDM requires a key-id to key mapping.  HLS doesn't
     // provide a key ID anywhere.  So although we could use the 'URI' attribute
     // to fetch the actual 16-byte key, without a key ID, we can't provide this
-    // automatically to the ClearKey CDM.  Instead, the application will have
-    // to use player.configure('drm.clearKeys', { ... }) to provide the key IDs
+    // automatically to the ClearKey CDM. By default we assume that keyId is 0,
+    // but we will try to get key ID from Init Segment.
+    // If the application want override this behavior will have to use
```

Comment:
I wonder if it would be worth putting something into the manifest update that gets the drm key overrides, so that if they are present we can avoid making a network request entirely.

Reply:
Maybe something similar to AES-128, I'll work on it

Reply:
I just noticed I made a typo last night. I meant "into the manifest parser interface"

Reply:
I added a cache!